dance/transforms/filter.py

```diff
@@ -884,3 +886,192 @@ def gini_func(x, weights=None):
     G_RSV = G_num / mean_RSV
 
     return G_RSV
+
+
+@register_preprocessor("filter", "gene")
+class FilterGenesScanpyOrder(BaseTransform):
+    """Scanpy filtering gene transformation with additional options.
+
+    Parameters
+    ----------
+    order
+        Order of (min_counts,min_cells,max_counts,max_cells),e.g["min_counts","min_cells","max_counts","max_cells"]
+    min_counts
+        Minimum number of counts required for a gene to be kept.
+    min_cells
+        Minimum number (or ratio) of cells required for a gene to be kept.
+    max_counts
+        Maximum number of counts required for a gene to be kept.
+    max_cells
+        Maximum number (or ratio) of cells required for a gene to be kept.
+    split_name
+        Which split to be used for filtering.
+    channel
+        Channel to be used for filtering.
+    channel_type
+        Channel type to be used for filtering.
+
+    """
+
+    def __init__(self, order: list, min_counts: Optional[int] = None, min_cells: Optional[Union[float, int]] = None,
+                 max_counts: Optional[int] = None, max_cells: Optional[Union[float, int]] = None,
+                 split_name: Optional[str] = None, channel: Optional[str] = None, channel_type: Optional[str] = "X",
+                 **kwargs):
+        super().__init__(**kwargs)
+        self.filter_genes_order = order
+        self.logger.info(f"choose filter_genes_order f{self.filter_genes_order}")
+        geneParameterDict = {
+            "min_counts": min_counts,
+            "min_cells": min_cells,
+            "max_counts": max_counts,
+            "max_cells": max_cells
+        }
+        if not set(order).issubset(set(geneParameterDict.keys())):
+            raise KeyError(f"An order should be in {geneParameterDict.keys()}")
+        self.geneScanpyOrderDict = {}
+        for key in geneParameterDict.keys():
+            if key in self.filter_genes_order:
+                self.geneScanpyOrderDict[key] = FilterGenesScanpy(**{key:
+                                                                     geneParameterDict[key]}, split_name=split_name,
+                                                                  channel=channel, channel_type=channel_type, **kwargs)
+            else:
+                self.logger.warning(f"{key} not in order,It makes no sense to set {key}")
+
+    def __call__(self, data: Data):
+        for parameter in self.filter_genes_order:
+            geneScanpyOrder = self.geneScanpyOrderDict[parameter]
+            geneScanpyOrder(data)
+
+
+@register_preprocessor("filter", "gene")
+class HighlyVariableGenesRawCount(BaseTransform):
```

Comment:
hvg is divided into three situations, namely flavor='seurat_v3', using n_top_genes and using mean and disp